spec/controllers/course_summaries_controller_spec.rb

```diff
@@ -46,6 +49,33 @@
         end
         expect(response.status).to eq(200)
       end
+
+      context 'tests the second csv row which contains out of values' do
+        def is_numbery(str)
+          str.to_i.to_s == str || str.to_f.to_s == str
+        end
+
+        it 'ensures the row contains the right number of integer/floating point values' do
```

Comment:
You can get rid of this test (and the helper function above), it's not super important. However, you *should* add a test that has a mixture of both assignments and grade entry forms, and make sure that the entry order is correct (sorted by id).